tests/integration/test_mapping.py

```diff
@@ -0,0 +1,159 @@
+from collections.abc import Iterable
+from itertools import chain
+from pathlib import Path
+import asyncio
+
+import pytest
+import pytest_asyncio
+
+from adcm_aio_client.core.client import ADCMClient
+from adcm_aio_client.core.filters import Filter
+from adcm_aio_client.core.mapping.refresh import apply_local_changes, apply_remote_changes
+from adcm_aio_client.core.mapping.types import MappingPair
+from adcm_aio_client.core.objects.cm import Bundle, Cluster, Host
+from tests.integration.bundle import pack_bundle
+from tests.integration.conftest import BUNDLES
+
+pytestmark = [pytest.mark.asyncio]
+
+type FiveHosts = tuple[Host, Host, Host, Host, Host]
+
+
+def build_name_mapping(*iterables: Iterable[MappingPair]) -> set[tuple[str, str, str]]:
+    return {(c.service.name, c.name, h.name) for c, h in chain.from_iterable(iterables)}
+
+
+@pytest_asyncio.fixture()
+async def cluster_bundle(adcm_client: ADCMClient, tmp_path: Path) -> Bundle:
+    bundle_path = pack_bundle(from_dir=BUNDLES / "complex_cluster", to=tmp_path)
+    return await adcm_client.bundles.create(source=bundle_path, accept_license=True)
+
+
+@pytest_asyncio.fixture()
+async def hostprovider_bundle(adcm_client: ADCMClient, tmp_path: Path) -> Bundle:
+    bundle_path = pack_bundle(from_dir=BUNDLES / "simple_hostprovider", to=tmp_path)
+    return await adcm_client.bundles.create(source=bundle_path, accept_license=True)
```

Comment:
`simple_hostprovider` does not have a license. Isn't it must raise an error?

Reply:
No, license acceptance is safe, added in here, because if it'll appear, it shouldn't break the tests